docs/cram-tests/index.md

```diff
@@ -0,0 +1,373 @@
+[sandbox](https://dune.readthedocs.io/en/stable/tests.html#sandboxing). What
+goes into the sandbox directory is determined by the value of the `cram/deps`
+field in your `dune` file. Let's write an explorative cram test that allows us
+to peek inside the sandbox. First, install the [tree-node-cli
+package](https://www.npmjs.com/package/tree-node-cli):
+
+```shell
+npm install --save-dev tree-node-cli
+```
+
+Then add a new cram test to `src/order-confirmation/tests.t`:
+
+```cram
+See all the files inside the sandbox
+  $ npx tree
+```
+
+Now when you run `npm run test`, you'll see some output like this:
+
+```diff
+@@ -24,3 +24,8 @@ Sandwich tests
+
+ See all the files inside the sandbox
+   $ npx tree
++  cram-tests
++  └── output
++      └── src
++          └── cram-tests
++              └── SandwichTests.mjs
+```
+
+The sandbox directory only contains a single file
+`output/src/cram-tests/SandwichTests.mjs`. Add another explorative cram test:
+
+```cram
+Show detail of SandwichTests.mjs
+  $ ls -la ./output/src/cram-tests/SandwichTests.mjs
+```
+
+The output should now look like this:
+
+```diff
+@@ -24,6 +24,12 @@ Sandwich tests
+
+ See all the files inside the sandbox
+   $ npx tree
++  cram-tests
++  └── output
++      └── src
++          └── cram-tests
++              └── SandwichTests.mjs
+
+ Show detail of SandwichTests.mjs
+   $ ls -la ./output/src/cram-tests/SandwichTests.mjs
++  lrwxr-xr-x  1 fhsu  staff  86 Mar 16 22:09 ./output/src/cram-tests/SandwichTests.mjs -> ../../../../../../../../default/src/cram-tests/output/src/cram-tests/SandwichTests.mjs
+```
+
+From this, you can see that `./output/src/cram-tests/SandwichTests.mjs` is
+actually a symbolic link which links to the real file inside your build
+directory. This explains why you can successfully run the test even when the
+rest of the `.mjs` files aren't present in the sandbox directory.
+
+## Better dependencies
+
+There's a problem with the cram test dependencies. The cram test is re-run if
+`SandwichTests.re` changes, but if `Item.re`, which `SandwichTests.re` depends
+on, were to change, it wouldn't trigger a test re-run. To see this, make the
+following change to `Item.Sandwich.toEmoji`:
+
+```reason
+let toEmoji = t =>
+  Printf.sprintf(
+    {js|🥪(%s)|js},
+    switch (t) {
+    | Portabello => {js|🍄|js}
+    | Ham => {js|🐷|js}
+    | Unicorn => {js|🦄|js}
+    | Turducken => {js|🦃🦆🐓|js} // [!code --]
+    | Turducken => {js|🦃🦆🐓💩|js} // [!code ++]
+    },
+  );
+```
+
+After running `npm run test`, you would expect a failing test, but that doesn't
+happen. You have to add `(alias melange)` to `cram/deps`:
+
+```dune
+(cram
+ (deps
+  (alias melange)
+  ./output/src/cram-tests/SandwichTests.mjs))
+```
+
+This adds the `Item.re` to the sandbox, but it doesn't cause the test to fail.
+The proper fix is to instead add `(alias melange)` to `cram/deps`:
+
+```dune
+(cram
+ (deps
+  (alias melange)
+  ./output/src/cram-tests/SandwichTests.mjs))
+```
```

Comment:
I'm not sure I get this, isn't the snippet exactly the same as the one above?

Reply:
Removed this extraneous section in https://github.com/melange-re/melange-for-react-devs/pull/30/commits/d898b7d92a9e445b71a149f487a306cbdb348236